### PR TITLE
check whether completion starts with quotation marks

### PR DESCRIPTION
**Pull request by chipshort**
This should eliminate some more false positives

### CODE REVIEW

**Comment by chipshort** [comment]
@Neverbirth I completely forgot to push the second commit. It is now using the scintilla styles to determine if the current position is within an attribute.

File: Plugins/DSLCompletion/DSLComplete.cs [hunk 1]
```diff
@@ -255,7 +255,8 @@ private string FindSelection(Check isAllowedChar, bool checkRight = true)
             
             var text = sci.GetLine(sci.CurrentLine);
 
-            if (!text.Contains("\"")) return null;
+            var style = sci.BaseStyleAt(sci.CurrentPos);
+            if (style != 6 && style != 7) return null;
```
Comment (Neverbirth) [inline Plugins/DSLCompletion/DSLComplete.cs]: Note that you may prefer to use `XML.DOUBLESTRING` and `XML.SINGLESTRING` from `ScintillaNet.Lexers`. Although FD codebase is full of just using the numeric value...

File: Plugins/DSLCompletion/DSLComplete.cs [hunk 2]
```diff
@@ -300,11 +301,14 @@ private string FindSelection(Check isAllowedChar, bool checkRight = true)
                 }
             }
 
+            if (sci.GetTextRange(left - 1, left) != "\"")
+                return null;
```
Comment (Neverbirth) [inline Plugins/DSLCompletion/DSLComplete.cs]: I guess now you would like to remove this if?

File: Plugins/DSLCompletion/DSLComplete.cs (hunk 1 above)
Comment (chipshort) [inline Plugins/DSLCompletion/DSLComplete.cs]: Good point, I didn't know about these.

**Pull request merged**